Move appearance enum to BLEProtocol

andresag01 opened the pull request:
There were two different enums (in GapAdvertisingData.h and blecommon.h) that
defined the appearance values for the BLE Appearance characteristic. This code
is removed and a single copy of the enum is included in BLEProcotol.h to
continue the effort of putting general information defined by the BLE spec in
a single file.
@rgrover @LiyouZhou @pan-

andresag01 commented:
**NOTE:** I will be submitting a pull request for ble-nrf51822 soon! Also, this pull request is related to IOTSFW-1373

rgrover commented on ble/BLEProtocol.h:
```diff
@@ -44,6 +44,67 @@ namespace BLEProtocol {
+            OUTDOOR_LOCATION_AND_NAVIGATION_POD            = 5188   /**< Outdoor Location and Navigation Pod. */
+        };
+    };
+    typedef AppearanceType::Type AppearanceType_t; /**< Alias for AppearanceType::Type */
```
use a @ref in the documentation so that doxygen automatically links the two concepts. `Alias for @ref AppearanceType::Type`

rgrover commented on ble/GapAdvertisingData.h:
```diff
@@ -131,71 +132,21 @@ class GapAdvertisingData
+    /**
+     * Appearance-type for BLEProtocol addresses.
+     *
+     * @note: deprecated. Use BLEProtocol::AppearanceType_t instead.
```
we should start using @refs. so change it to `@note: deprecated. Use @ref BLEProtocol::AppearanceType_t instead.`

rgrover commented:
hope you've ensured that the two enums were identical.

andresag01 commented:
I noticed that some of the examples in ble-examples use the appearance values. Once everyone give their +1 to this pull request I will proceed to update those to ensure that everything works are expected.

andresag01 commented:
@rgrover: I think they are identical, but I think I will better check the actual BLE spec once more and will get back to you soon.

rgrover commented:
@andresag01 if applications need to be changed, then this becomes an API breaking change. I'd like to avoid that.
We should retain a deprecated version of the original enum so that applications don't need to be changed. At the same time we'll update our examples to reflect the new type.

andresag01 commented:
@rgrover: I have addressed all comments and now the API is compatible with older versions of code. Also I found that we were missing the GENERIC_WIGHT_SCALE value from the BLE specification. Finally, I noticed that when using the deprecated enums GCC generates the following warning:
```
 warning: large integer implicitly truncated to unsigned type [-Woverflow]
     ble.gap().accumulateAdvertisingPayload(GapAdvertisingData::GENERIC_TAG);
```

pan- commented:
c++ in all its glory!

You have to add a type for the deprecated enum in Gap.h and an overload of accumulateAdvertisingPayload for it.

andresag01 commented:
closing this pull request because it causes a breaking change.
